src/main/java/cart/domain/product/ProductService.java

```diff
@@ -24,42 +23,48 @@ public ProductService(final ProductDao productDao) {
         this.productDao = productDao;
     }
 
-    public List<ProductDto> getProducts() {
-        final List<Product> products = productDao.findAll();
-        return products.stream()
-                .map(product -> new ProductDto(product.getId(), product.getName(), product.getImageUrl(),
-                        product.getPrice(), product.getCategory())).collect(Collectors.toList());
+    public List<Product> getProducts() {
+        return productDao.findAll();
     }
 
-    public Long save(final ProductDto productDto) {
-        return productDao.insert(productDto.toEntity());
+    public Long save(final ProductRequest productRequest) {
+        final Product product = map2Product(productRequest);
+        return productDao.insert(product);
     }
 
-    public void update(final Long id, final ProductDto productDto) {
-        int updatedCount = productDao.update(productDto.toEntity());
+    public void update(final Long id, final ProductRequest productRequest) {
+        final Product product = map2Product(productRequest);
+        int updatedCount = productDao.update(id, product);
         if (updatedCount != 1) {
             throw new GlobalException(ErrorCode.PRODUCT_NOT_FOUND);
         }
     }
 
+    private Product map2Product(final ProductRequest productRequest) {
+        return new Product(
+                productRequest.getName(),
+                productRequest.getImageUrl(),
+                productRequest.getPrice(),
+                productRequest.getCategory());
+    }
+
     public void delete(final Long id) {
         int deletedCount = productDao.deleteById(id);
         if (deletedCount == 0) {
             throw new GlobalException(ErrorCode.PRODUCT_NOT_FOUND);
         }
         if (deletedCount > 1) {
-            log.error("error = {}", "delete count is more than 2");
+            log.error("error = {}", "삭제 상황에서 중복된 상품이 존재합니다. DB를 확인하세요.");
             throw new GlobalException(ErrorCode.INVALID_DELETE);
         }
     }
 
-    public ProductDto getById(final Long id) {
+    @Transactional(readOnly = true)
+    public Product getById(final Long id) {
         final Optional<Product> productById = productDao.findById(id);
         if (productById.isEmpty()) {
             throw new GlobalException(ErrorCode.PRODUCT_NOT_FOUND);
         }
```

Comment:
이 부분은 1단계에서 못보고 넘어갔었군요 😱 
아래와 같이 구현해보는건 어떠신가요?
```suggestion
        productById.orElseGet(() -> new GlobalException(ErrorCode.PRODUCT_NOT_FOUND))
```

Reply:
아하.. 제가 놓쳤습니다! 감사합니다!

Reply:
앗 orElseThrow 였습니다!!